Material3로 migration

easyhooon opened the pull request:
- 프로젝트에서 사용하는 라이브러리 version update
- Material3 관련 의존성 추가 
- Theme.MaterialCompontents -> Theme.Material3

제가 알기론 Material 3로 이전하면 컴포넌트들이 필요로하는 파라미터들이 많이 바뀌는걸로 아는데 Theme를 변경하였는데도 그대로 빌드가 되는군요...
com.googl.android:material:material 의존성 자체를 제거 해버려야 반영되는지 해서 지워봤는데 이리면 Theme.Material3를 찾을 수 없다는 에러가 발생해서 일단 material3 관련 의존성 추가해서 PR 올립니다.

easyhooon committed:
library version update

easyhooon committed:
Theme Material3 로 변경

easyhooon commented:
그리고 혹시 gitignore 에 파일들 몇개 추가해도 될까요? 깃에서 추적하지 않아도 될 파일들이 몇개 있어가지구
<img width="360" alt="image" src="https://github.com/mwy3055/MyVoca/assets/51016231/8273b78c-d96d-4302-b2d5-d0581760c5aa">

mwy3055 commented:
빌드가 되는 이유는 여전히 M2로 빌드되고 있기 때문입니다.

![image](https://github.com/mwy3055/MyVoca/assets/45386920/8631c42e-d2c9-47ae-8fe2-d4b6ad4682fb)

`com.google.android.material:material`은 View 시스템에서의 머티리얼 컴포넌트를 담고 있어서 우리 앱과는 무관합니다(사실 필요없어 보입니다).

현재 코드에서 import된 `androidx.compose.material` 의존성을 모두 `androidx.compose.material3`으로 바꾸고, 매개변수 이름 등 변경점을 대응해야 합니다.

mwy3055 reviewed (changes requested):
빌드가 되는 이유는 여전히 M2로 빌드되고 있기 때문입니다.

![image](https://github.com/mwy3055/MyVoca/assets/45386920/8631c42e-d2c9-47ae-8fe2-d4b6ad4682fb)

`com.google.android.material:material`은 View 시스템에서의 머티리얼 컴포넌트를 담고 있어서 우리 앱과는 무관합니다(사실 필요없어 보입니다).

현재 코드에서 import된 `androidx.compose.material` 의존성을 모두 `androidx.compose.material3`으로 바꾸고, 매개변수 이름 등 변경점을 대응해야 합니다.

[공식 문서](https://developer.android.com/jetpack/compose/designsystems/material2-material3#approaches)를 참고하시면 도움이 될 수도..?

mwy3055 commented:
> 그리고 혹시 gitignore 에 파일들 몇개 추가해도 될까요? 깃에서 추적하지 않아도 될 파일들이 몇개 있어가지구 <img alt="image" width="360" src="https://user-images.githubusercontent.com/51016231/278568567-8273b78c-d96d-4302-b2d5-d0581760c5aa.png">

좋습니다

mwy3055 commented:
Color, Theme은 이 파일을 적용해 주세요 (README는 필요 없습니다)

[material-theme.zip](https://github.com/mwy3055/MyVoca/files/13197336/material-theme.zip)

mwy3055 commented:
아 그리고 Gradle 버전도 8.0으로 올려주시겠어요? 여기저기서 에러가 너무 많이 나네요

easyhooon committed:
gitignore update

easyhooon committed:
gitignore update 반영

easyhooon commented:
> 아 그리고 Gradle 버전도 8.0으로 올려주시겠어요? 여기저기서 에러가 너무 많이 나네요

gradle-wrapper.properties 파일 커밋을 누락했네요 반영하겠습니다

easyhooon committed:
BaselineProfileGenerator 실행 관련 에러 해결

AGP 8.1 버전에 맞는 코드로 변경 및 proguard-rules.pro 파일 수정

easyhooon commented:
[BaselineProfileGenerator 실행 관련 에러 해결](https://github.com/mwy3055/MyVoca/pull/140/commits/877d00e94fc9ca9d96e62c4dc835da94fe366f1b)

AGP 버전을 업데이트 했을 경우 기존 코드와 호환이 되지 않아 아래의 문서들을 참고해서 커밋했습니다

reference)
https://developer.android.com/topic/performance/baselineprofiles/create-baselineprofile?hl=ko
https://singo112ok.tistory.com/229

easyhooon committed:
BackdropScaffold 를 제외한 Component, Screen Material3 로 migration

BackdropScaffold 는 Material3 에서는 아직 사용할 수 없는 API 이므로 다른 Component 로 대체 해야함
-> UI 가 변경 되었으므로 이를 ModalBottomSheet 로 구현할 예정

mwy3055 committed:
gradle: jacoco 업데이트 대응

mwy3055 committed:
Merge remote-tracking branch 'origin/refactor/m3-migration' into refactor/m3-migration

mwy3055 reviewed (approved):
확인~ 다음 PR부터는 오류나는 곳 없는지 한번 더 확인해 주세요

easyhooon commented:
> 확인~ 다음 PR부터는 오류나는 곳 없는지 한번 더 확인해 주세요

넵 ㅜ test 까지 전부 빌드 되는지 확인하도록 하겠습니다
